### PR TITLE
(maint) Unvendor leatherman (based on facter's unvendoring)

### PR DESCRIPTION
**Pull request by kylog**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,10 +18,30 @@ endif()
...
 
-# Leatherman setup
-list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/vendor/leatherman/cmake")
+include(FeatureSummary)
+
+SET(LEATHERMAN_COMPONENTS locale catch nowide logging util file_util dynamic_library execution ruby rapidjson)
+
+# We look for curl early, because whether or not we link to the leatherman curl library
+# is dependent on whether or not we find curl on the system.
+if ((("${CMAKE_SYSTEM_NAME}" MATCHES "Linux") OR WIN32) AND NOT WITHOUT_CURL)
```
Comment (MikaelSmith) [inline CMakeLists.txt]: Restricting to Linux or WIN32 doesn't make sense for all projects, just Facter (where those are the only OS we expect to run on EC2 or GCE). I'd change it to just `NOT WITHOUT_CURL`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,10 +18,30 @@ endif()
 
 enable_testing()
 
-# Leatherman setup
-list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/vendor/leatherman/cmake")
+include(FeatureSummary)
+
+SET(LEATHERMAN_COMPONENTS locale catch nowide logging util file_util dynamic_library execution ruby rapidjson)
```
Comment (MikaelSmith) [inline CMakeLists.txt]: Project doesn't need all of these. If we're just including them as an example, add a comment to reflect that and mention which are actually used by the project.

**Comment by kylog** [comment]
Fixing that and a couple other things I just noticed. Stand by ....

**Comment by kylog** [comment]
I think this is ready for review now.

**Comment by MikaelSmith** [comment]
:+1:

**Pull request merged**